base_export_email/ir_actions.py

```diff
@@ -0,0 +1,157 @@
+                    'email.template', string='Template', ondelete='restrict'),
+                'saved_export_id': fields.many2one(
+                    'ir.exports', string='Saved export', ondelete='restrict'),
+                'fields_to_export': fields.text(
+                    'Fields to export',
+                    help="The list of fields to export. \
+                          The format is ['name', 'seller_ids/product_name']")
+                }
+
+    _defaults = {'fields_to_export': '[]'}
+
+    def onchange_model_id(self, cr, uid, ids, model_id, context=None):
+        """
+        Used to set correct domain on filter_id and saved_export_id
+        """
+        data = {'model': False,
+                'filter_id': False,
+                'email_template_id': False,
+                'saved_export_id': False}
+        if model_id:
+            model = self.pool['ir.model'].browse(cr, uid, model_id,
+                                                 context=context)
+            data.update({'model': model.model})
+        return {'value': data}
+
+    def _search_data(self, cr, uid, action, context=None):
+        """
+         search data to export
+        """
+        obj_pool = self.pool[action.model]
+        domain = action.filter_id and safe_eval(
+            action.filter_id.domain, {'time': time}) or []
+        ctx = action.filter_id and safe_eval(
+            action.filter_id.context) or context
+        return obj_pool.search(cr, uid, domain,
+                               offset=0, limit=False, order=False,
+                               context=ctx)
+
+    def _export_data(self, cr, uid, obj_ids, action, context=None):
+        """
+        export data
+        """
+        obj_pool = self.pool[action.model]
+        export_fields = []
+        if action.fields_to_export:
+            export_fields = safe_eval(action.fields_to_export)
+        if action.saved_export_id:
+            # retrieve fields of the selected list
+            export_fields.extend(
+                [efl.name for efl in
+                 action.saved_export_id.export_fields])
+        return export_fields, obj_pool.export_data(
+            cr, uid, obj_ids, export_fields,
+            context=context).get('datas', [])
+
+    def _send_email(self, cr, uid, action, export_fields, export_data,
+                    context=None):
+        """
+        Prepare a message with the exported data to send with the
+        template of the configuration
+        """
+        mail_compose = self.pool['mail.compose.message']
+        values = mail_compose.onchange_template_id(
+            cr, uid, 0, action.email_template_id, 'comment',
+            action.model, 0, context=context)['value']
+        values['partner_ids'] = [
+            (4, partner_id) for partner_id in values.pop('partner_ids',
+                                                         [])
+        ]
+        export = main.CSVExport()
+        filename = export.filename(action.model)
```

Comment:
Hooking on the web controller for a backend module does not look right. Is there not a better way?